packages/commonwealth/client/scripts/hooks/useSearchResults.ts

```diff
@@ -0,0 +1,66 @@
+import { useMemo } from 'react';
+import { useDebounce } from 'usehooks-ts';
+
+import { APIOrderBy, APIOrderDirection } from '../helpers/constants';
+import { useSearchThreadsQuery } from '../state/api/threads';
+import { useSearchCommentsQuery } from '../state/api/comments';
+import { useSearchChainsQuery } from '../state/api/chains';
+import { useSearchProfilesQuery } from '../state/api/profiles';
+import { SearchScope } from '../models/SearchQuery';
+import app from '../state';
+
+const NUM_RESULTS_PER_SECTION = 2;
+
+type Filter = 'threads' | 'replies' | 'communities' | 'members';
+
+const useSearchResults = (
+  searchTerm: string,
+  filters: Filter[],
+  resultsPerSection?: number
+) => {
+  const chainId = app.activeChainId() || 'all_chains';
+  const debouncedSearchTerm = useDebounce<string>(searchTerm, 500);
+
+  const sharedQueryOptions = {
+    chainId,
+    searchTerm: debouncedSearchTerm,
+    limit: resultsPerSection ? resultsPerSection : NUM_RESULTS_PER_SECTION,
+    orderBy: APIOrderBy.Rank,
+    orderDirection: APIOrderDirection.Desc,
+  };
+  const queryEnabled = debouncedSearchTerm.length > 0;
+
+  const { data: threadsData } = useSearchThreadsQuery({
+    ...sharedQueryOptions,
+    enabled: queryEnabled,
+  });
+
+  const { data: commentsData } = useSearchCommentsQuery({
+    ...sharedQueryOptions,
+    enabled: queryEnabled,
+  });
+
+  const { data: chainsData } = useSearchChainsQuery({
+    ...sharedQueryOptions,
+    enabled: queryEnabled,
+  });
+
+  const { data: profilesData } = useSearchProfilesQuery({
+    ...sharedQueryOptions,
+    includeRoles: false,
+    enabled: queryEnabled,
+  });
+
+  const searchResults = useMemo(() => {
+    return {
+      [SearchScope.Threads]: filters.includes("threads") ? threadsData?.pages?.[0]?.results || [] : [],
+      [SearchScope.Replies]: filters.includes("replies") ? commentsData?.pages?.[0]?.results || [] : [],
+      [SearchScope.Communities]: filters.includes("communities") ? chainsData?.pages?.[0]?.results || [] : [],
+      [SearchScope.Members]: filters.includes("members") ? profilesData?.pages?.[0]?.results || [] : [],
+    };
+  }, [threadsData, chainsData, profilesData, commentsData, filters]);
```

Comment:
Here, it's fetching __all__ results (threads, comments, chains, profiles), but then returning only some results, based on filter.

For efficiency, the `enabled` option should be set based on the filter, so that it's not fetching data that won't be returned/rendered.

Reply:
[Addressed](https://github.com/hicommonwealth/commonwealth/commit/046a7bc5daa2fcc8a7bdf8c520ef80fe3ab8f6fa).